src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -190,7 +191,9 @@ private Type0CommandMessageSection<BsonDocument> CreateType0Section(ConnectionDe
             var dbElement = new BsonElement("$db", _databaseNamespace.DatabaseName);
             extraElements.Add(dbElement);
 
-            if (_readPreference != null && _readPreference != ReadPreference.Primary)
+            if (connectionDescription.IsMasterResult.ServerType != ServerType.Standalone
+                && _readPreference != null
+                && _readPreference != ReadPreference.Primary)
```

Comment:
It's good that only a very small change was required to implement this.

But it's funny how hard this small change tunred out to be to test!

Reply:
Agreed 😞